### PR TITLE
RUMM-812 Shopist: RUM Autoinstrumentation

### PR DESCRIPTION
**Pull request by buranmert**
### What and why?

We test RUM autoinstrumentation in Shopist app

### How?

What is autoinstrumented:

1. network requests
2. `tap` user actions
3. views

What is to be autoinstrumented later

1. custom and swipe user actions

batch-results

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference

### CODE REVIEW

**Comment by buranmert** [comment]
Resources volume was low due to caching
Now fixed, retrying... ⏳

File: Shopist/Shopist/Scenes/CheckoutViewController.swift [hunk 1]
```diff
@@ -67,7 +66,6 @@ internal final class CheckoutViewController: UITableViewController {
 
     override func viewWillAppear(_ animated: Bool) {
```
Comment (ncreated) [inline Shopist/Shopist/Scenes/CheckoutViewController.swift:67]: We should use `viewDidAppear()`, to get the correct resources assignment.

File: Shopist/Shopist/Scenes/ListViewController.swift [hunk 2]
```diff
@@ -27,7 +26,6 @@ internal class ListViewController: UICollectionViewController {
 
     override func viewWillAppear(_ animated: Bool) {
```
Comment (ncreated) [inline Shopist/Shopist/Scenes/ListViewController.swift:27]: Same: we should use `viewDidAppear()`, to get the correct resources assignment

File: Shopist/Shopist/Scenes/ProductViewController.swift [hunk 3]
```diff
@@ -34,15 +34,9 @@ internal class ProductViewController: UIViewController {
 
     override func viewWillAppear(_ animated: Bool) {
```
Comment (ncreated) [inline Shopist/Shopist/Scenes/ProductViewController.swift:35]: Same: we should use `viewDidAppear()`, to get the correct resources assignment

**Review by ncreated — CHANGES_REQUESTED** [review]
Great that we get more with less code 👍. The reason of resources volume drop may be because of starting requests in `viewWillAppear()`. Let's switch to `viewDidAppear()` as pointed out and it should be corrected. Once we improve this in SDK, we can switch to "will" again.

**Comment by buranmert** [comment]
after-v3

now looks good ✅

**Review by ncreated — APPROVED** [review]

**Pull request merged**